### ISSUE TITLE
Allow conditional registration/recognition of templates and template parts

### ISSUE THREAD

**Issue by webmandesign** [WordPress/gutenberg#51987]
## What problem does this address?

In my block theme I am defining WooCommerce templates, and WooCommerce template parts.

I'd like to ask, whether it is somehow possible to **remove these templates and template parts** from Site Editor **when WooCommerce is not active**.

Also, the templates are being registered as custom templates and WordPress offers them in page/post edit screen, which is not desirable.

Currently all of the above is available to user even if WooCommerce is inactive.

## What is your proposed solution?

It would be great to allow a template and a template part registration/recognition conditionally.

Basically, I'd like to get rid of all of these when WooCommerce is not active:

![wc-templates](https://github.com/WordPress/gutenberg/assets/2456735/b4ef0476-075c-4645-859b-dfaaefff79da)
![wc-template-parts](https://github.com/WordPress/gutenberg/assets/2456735/3cdc9f51-34e7-4a6c-b43d-52927198e99d)
![wc-templates-custom](https://github.com/WordPress/gutenberg/assets/2456735/5fc1432b-70ea-4ce4-b0f6-bbf686e61f9f)

**Comment by skorasaurus** [comment]
Hi, 

Thanks for reporting; 
This has been discussed and/or requested in https://github.com/WordPress/gutenberg/issues/29076 . Feel free to [add your reaction](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-conversations-on-github#reacting-to-ideas-in-comments) and follow its progress in the issue.

Please reopen if you believe that this issue is not similar to the issued mentioned above.
